crates/rt/src/thread.rs

```diff
@@ -16,18 +15,18 @@ pub fn wait_for_completion() {
         );
     }
 
-    const MAX_THREADS: usize = 32;
     let cur_thread_id = unsafe { sys::furi_thread_get_current_id() };
     let app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(cur_thread_id)) };
-    let mut thread_ids: [sys::FuriThreadId; MAX_THREADS] = [ptr::null_mut(); MAX_THREADS];
+    let thread_list = unsafe { sys::furi_thread_list_alloc() };
 
-    'outer: loop {
-        let thread_count =
-            unsafe { sys::furi_thread_enumerate(thread_ids.as_mut_ptr(), MAX_THREADS as u32) }
-                as usize;
+    unsafe { sys::furi_thread_enumerate(thread_list) };
+    let n_threads = unsafe { sys::furi_thread_list_size(thread_list) };
 
-        for &thread_id in thread_ids[..thread_count].iter() {
-            let thread_app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(thread_id)) };
+    'outer: loop {
+        for i in 0..n_threads {
+            let item = unsafe { furi_thread_list_get_at(thread_list, i) };
+            let thread_id = unsafe { furi_thread_get_id((*item).thread) };
+            let thread_app_id = unsafe { CStr::from_ptr((*item).app_id) };
 
             if thread_id == cur_thread_id || thread_app_id != app_id {
```

Comment:
I suggest that we only lazily initialize `thread_app_id` if the first check (`thread_id == cur_thread_id`) lets us pass.

This is important since `CStr::from_ptr(..)` is $O(n)$

Reply:
Good idea. Moved into two separate checks.